exasol/ds/sandbox/runtime/ansible/roles/entrypoint/files/entrypoint.py

```diff
@@ -28,25 +30,73 @@ def arg_parser():
         "--jupyter-server", metavar="<PATH-TO-JUPYTER-BINARY>",
         help="start server for Jupyter notebooks",
     )
+    parser.add_argument(
+        "--user", type=str,
+        help="user name for running jupyter server",
+    )
+    parser.add_argument(
+        "--jupyter-logfile", type=Path,
+        help="path to write Jupyter server log messages to",
+    )
     parser.add_argument(
         "--warning-as-error", action="store_true",
         help="treat warning as error",
     )
     return parser
 
 
-def start_jupyter_server(binary_path: str, notebook_dir: str):
-    subprocess.run([
+def start_jupyter_server(
+        binary_path: str,
+        notebook_dir: str,
+        logfile: Path,
+        user: str,
+        poll_sleep: float = 1,
+):
+    """
+    :param poll_sleep: specifies the waiting time in seconds before reading
+    the a line from the logfile.
+    """
+    def exit_on_error(rc):
+        if rc is not None and rc != 0:
+            log_messages = logfile.read_text()
+            print(
+                f"Jupyter Server terminated with error code {rc},"
+                f" Logfile {logfile} contains:\n{log_messages}",
+                flush=True,
+            )
+            sys.exit(rc)
+
+    command_line = [
         binary_path,
         f"--notebook-dir={notebook_dir}",
         "--no-browser",
         "--allow-root",
-    ])
-    print(cleandoc(
-        """
-        Server for Jupyter has been started.
-        You can connect with https://localhost:8888
-        """))
+    ]
+    with open(logfile, "w") as f:
+        p = subprocess.Popen(command_line, stdout=f, stderr=f)
+
+    success_message = cleandoc(f"""
+        Server for Jupyter has been started successfully.
+        You can connect with https://<ip address>:8888.
+        If using a Docker daemon on your local machine then ip address is localhost.
+
+        ┬ ┬┌─┐┌┬┐┌─┐┌┬┐┌─┐  ┬ ┬┌─┐┬ ┬┬─┐   ┬┬ ┬┌─┐┬ ┬┌┬┐┌─┐┬─┐  ┌─┐┌─┐┌─┐┌─┐┬ ┬┌─┐┬─┐┌┬┐ ┬
+        │ │├─┘ ││├─┤ │ ├┤   └┬┘│ ││ │├┬┘   ││ │├─┘└┬┘ │ ├┤ ├┬┘  ├─┘├─┤└─┐└─┐││││ │├┬┘ ││ │
+        └─┘┴  ─┴┘┴ ┴ ┴ └─┘   ┴ └─┘└─┘┴└─  └┘└─┘┴   ┴  ┴ └─┘┴└─  ┴  ┴ ┴└─┘└─┘└┴┘└─┘┴└──┴┘ o
+
+        To update the password as user {user} run
+            {binary_path} server <new password>
+    """)
+    with open(logfile, "r") as f:
+        regexp = re.compile("Jupyter Server .* is running at:")
+        while True:
+            time.sleep(poll_sleep)
```

Comment:
you don't need a sleep here,, instead of poll you can use wait with a timeout

Reply:
In line 98 we do not expect the subprocess to terminate.
A timeout is also inappropriate, as this raises an error if the subprocess does not terminate. 
The `sleep` in the current line is only to slow down the polling frequency. 
It would be possible to remove the `sleep` but the busy waiting would be more busy, and I would prefer to avoid unnecessary load.

Reply:
ok, I had the wait wrongly in memory